chainer/__init__.py

```diff
@@ -4,16 +4,26 @@
 import threading
 import warnings
 
+from chainer import cuda  # NOQA
+from chainer import dataset  # NOQA
+from chainer import datasets  # NOQA
 from chainer import flag
 from chainer import function
 from chainer import function_set
+from chainer import functions  # NOQA
```

Comment:
I think, in most cases, user uses optimizers and serializers only once in their scripts, so it is convenient to import them automatically so that users do not have to write an import line for them. I also think that it is confusing to only import some sub modules while leave others, so it is better to also import functions and links.